applications/FluidDynamicsApplication/custom_elements/data_containers/two_fluid_fractional_navier_stokes/vectorial_convection_fractional_element_data.h

```diff
@@ -0,0 +1,127 @@
+///@{
+
+///@name Kratos classes
+///@{
+
+template< size_t TDim, size_t TNumNodes >
+class VectorialConvectionFractionalElementData : public FluidElementData<TDim,TNumNodes, true>
+{
+public:
+
+///@name Type Definitions
+///@{
+
+using NodalScalarData = typename FluidElementData<TDim,TNumNodes, true>::NodalScalarData;
+using NodalVectorData = typename FluidElementData<TDim,TNumNodes, true>::NodalVectorData;
+using ShapeFunctionsType = typename FluidElementData<TDim, TNumNodes, true>::ShapeFunctionsType;
+using ShapeDerivativesType = typename FluidElementData<TDim, TNumNodes, true>::ShapeDerivativesType;
+using MatrixRowType = typename FluidElementData<TDim, TNumNodes, true>::MatrixRowType;
+
+static constexpr std::size_t BlockSize = TDim + 1;
+///@}
+///@name Public Members
+///@{
+
+
+NodalVectorData Velocity_Fractional;
+NodalVectorData Velocity_OldStep1;
+NodalVectorData Velocity_OldStep2;
+NodalVectorData Velocity_OldStep3;  // Old step acceleration approximation using BDF2
+NodalVectorData MeshVelocity;
+
+double DynamicViscosity;
+double DeltaTime;		   // Time increment
+double DynamicTau;         // Dynamic tau considered in ASGS stabilization coefficients
+double bdf0;
+double bdf1;
+double bdf2;
+
+// Auxiliary containers for the symbolically-generated matrices
+BoundedMatrix<double,TNumNodes*(TDim),TNumNodes*(TDim)> lhs;
+array_1d<double,TNumNodes*(TDim)> rhs;
+
+double ElementSize;
+
+
+
+///@}
+///@name Public Operations
+///@{
+
+void Initialize(const Element& rElement, const ProcessInfo& rProcessInfo) override
+{
+
+
+    const Geometry< Node >& r_geometry = rElement.GetGeometry();
+    this->FillFromHistoricalNodalData(Velocity_OldStep1,VELOCITY,r_geometry,1);
+    this->FillFromHistoricalNodalData(Velocity_OldStep2,VELOCITY,r_geometry,2);
+    // this->FillFromHistoricalNodalData(Velocity_OldStep3, VELOCITY, r_geometry, 3); // Old step acceleration approximation using BDF2
+    this->FillFromHistoricalNodalData(Velocity_Fractional, FRACTIONAL_VELOCITY, r_geometry, 0);
+    this->FillFromProcessInfo(DeltaTime,DELTA_TIME,rProcessInfo);
+    this->FillFromProcessInfo(DynamicTau,DYNAMIC_TAU,rProcessInfo);
+    const Vector& BDFVector = rProcessInfo[BDF_COEFFICIENTS];
+    bdf0 = BDFVector[0];
+    bdf1 = BDFVector[1];
+    bdf2 = BDFVector[2];
+    noalias(lhs) = ZeroMatrix(TNumNodes*(TDim),TNumNodes*(TDim));
+    noalias(rhs) = ZeroVector(TNumNodes*(TDim));
+}
+
+void UpdateGeometryValues(
+    unsigned int IntegrationPointIndex,
+    double NewWeight,
+    const MatrixRowType& rN,
+    const BoundedMatrix<double, TNumNodes, TDim>& rDN_DX) override
+{
+    FluidElementData<TDim,TNumNodes, true>::UpdateGeometryValues(IntegrationPointIndex, NewWeight,rN,rDN_DX);
+    ElementSize = ElementSizeCalculator<TDim,TNumNodes>::GradientsElementSize(rDN_DX);
+}
+
+static int Check(const Element& rElement, const ProcessInfo& rProcessInfo)
+{
+    const Geometry< Node >& r_geometry = rElement.GetGeometry();
```

Comment:
```suggestion
    const auto& r_geometry = rElement.GetGeometry();
```
Please do so in similar ones.